Framework/DataHandling/src/LoadILLPolarizedDiffraction.cpp

```diff
@@ -0,0 +1,411 @@
+constexpr size_t NUMBER_MONITORS = 2;
+// The conversion factor from radian to degree
+constexpr double DEG_TO_RAD = M_PI / 180.;
+} // namespace
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_NEXUS_FILELOADER_ALGORITHM(LoadILLPolarizedDiffraction)
+
+/// Returns confidence. @see IFileLoader::confidence
+int LoadILLPolarizedDiffraction::confidence(NexusDescriptor &descriptor) const {
+
+  // fields existent only at the ILL Diffraction
+  if (descriptor.pathExists("/entry0/D7")) {
+    return 80;
+  } else {
+    return 0;
+  }
+}
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string LoadILLPolarizedDiffraction::name() const {
+  return "LoadILLPolarizedDiffraction";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int LoadILLPolarizedDiffraction::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string LoadILLPolarizedDiffraction::category() const {
+  return "DataHandling\\Nexus;ILL\\Diffraction";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string LoadILLPolarizedDiffraction::summary() const {
+  return "Loads ILL polarized diffraction nexus files.";
+}
+
+/**
+ * Constructor
+ */
+LoadILLPolarizedDiffraction::LoadILLPolarizedDiffraction()
+    : IFileLoader<NexusDescriptor>(), m_instNames({"D7"}) {}
+
+/**
+ * Initialize the algorithm's properties.
+ */
+void LoadILLPolarizedDiffraction::init() {
+  declareProperty(std::make_unique<FileProperty>("Filename", "",
+                                                 FileProperty::Load, ".nxs"),
+                  "File path of the data file to load");
+  declareProperty(std::make_unique<WorkspaceProperty<API::WorkspaceGroup>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "The output workspace.");
+  std::vector<std::string> positionCalibrationOptions{"None", "Nexus",
+                                                      "YIGFile"};
+  declareProperty(
+      "PositionCalibration", "None",
+      std::make_shared<StringListValidator>(positionCalibrationOptions),
+      "Select the type of pixel position calibration. If None, the pixel "
+      "positions are read from IDF file. If Nexus, the positions are read from "
+      "Nexus file. If YIGFile, then the calibration twotheta data is loaded "
+      "from a user-defined calibration file.");
+
+  declareProperty(std::make_unique<FileProperty>(
+                      "YIGFilename", "", FileProperty::OptionalLoad, ".xml"),
+                  "File path of the YIG calibration data file to load");
+  setPropertySettings("YIGFilename",
+                      std::make_unique<Kernel::EnabledWhenProperty>(
+                          "PositionCalibration", IS_EQUAL_TO, "YIGFile"));
+}
+
+/**
+ * Executes the algorithm.
+ */
+void LoadILLPolarizedDiffraction::exec() {
+
+  Progress progress(this, 0, 1, 2);
```

Comment:
This is very coarse progress reporting, which is fine as long as the files load quickly, otherwise, you probably want to pas the progress object down and report at a finer level.

The algorithm can only be canceled at a progress reporting point, so if they are more than a second apart that is too much.

Reply:
The largest files with TOF data load in about 1 s on a local machine, so perhaps the move of progress reporting to the function that actually reads from NeXuS is not necessary